### DISCUSSION TITLE
Question about the "index.html" file.

### DISCUSSION THREAD

**Discussion by KirikauKiwi** [DIY-Flow-Bench/DIY-Flow-Bench#229] [Code Development]
So a couple of times I have had an issue with the index.html file.

What I think is happening when changes in the code are saved, a build processed and then uploaded to the ESP32 board, I am noticing the index.html doesn't change and other times it does.

- When should we expect it to change, or how do we make it change consistently?
- When do we and when don't we need to delete the index.html file using the GUI and load the new index.html Guessing this is every time it is updated, so the bullet point above is more relevant.
- Have I been missing something when making changes in the code, that is preventing the index.html from updating. upon Build and Upload to the ESP32 board.

A couple of times I have had to clear the flash memory, I think @stefan63 has had to do the same.

**Comment by DeeEmm** [comment]
For the benefit of all + future reference

The index.html file is only uploaded via SPIFFS, not as part of the regular compile / upload process. So if the index.html file is changed as part of the code development, it needs to be compiled and uploaded separately.

The index file is compiled using 'GULP'. There are some instructions on how to set up your environment for GULP in the WIKI.

https://github.com/DeeEmm/DIY-Flow-Bench/wiki/WEB-UI

However, you do not need to compile it as I compile it and push the change out to the repro with each commit / PR

The compiled index.html file is in the /data folder

I generally tend to use VScode to upload the index.html file. But then my environment is set up for this. Normally it is just a case of building the filesystem image and then uploading the filesystem image

<img width="310" alt="Screenshot 2024-11-12 at 3 06 31 pm" src="https://github.com/user-attachments/assets/0a165eb9-00b1-4e0a-9e54-7b4f8cdcafcb">

This does require that the updated and compiled index.html file is present in the /data folder if you have not compiled it yourself

For all Commits / PRs where the compiled index.html file has been changed, the file will form part of the commit and will be updated. So when you are checking out the current version, the file should be updated in your local /data folder.

If you are not using GIT and are manually downloading the files from the repo then it is important to grab all files and folders, and not just the files at the root of the DIYFB-Flow-Bench folder.

If you are manually uploading via the GUI, then the existing file needs to be deleted first. This will then present the initial file upload page which will allow you to browse to to the file and upload it.

-----------------------

At the moment I am working on the index.html file versioning as currently there is no way for you to know if you need to update the index file, and there is also no way for me to know what index file you are currently running. This is obviously a bit of a nightmare for fault finding and diagnosis when there are frequent changes being made to the file

The standard advice is of course _always_ upload the index.html file and then you cannot go wrong.

-----------------------

I suspect that the issue you are experiencing is as a result of either not updating your local copy of the index file in your /data folder or using a file from a different source. Obviously this is just an assumption as I don't actually know what workflow you guys are using.

-----------------------

First stage of the index.html versioning is to automatically update the index file with the latest build number at the time it is compiled. This will be visible in the info panel. 

Then we can think about how this info can be used to determine if your existing index file is out of date. The easiest way is with a callback, but I'm not a great fan of such trickery.
